### PR TITLE
adding a generic service template

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Creating a generic service template.

Fixes # (issue)
Feature # (details)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor

## How Has This Been Tested?

Provide instructions so we can reproduce, and list any relevant details for your test configuration. Please mention if this is a breaking change which will impact consuming tool(s).

- [ ] Unit Test:
- [ ] E2E Test:
- [ ] Other Test:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: template/manifests/Service/draft.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - name: "GENERATORLABEL"
+    type: "string"
+    kind: "label"
+    description: "the label to identify who generated the resource"
+    default:
+      value: "draft"
+  - name: "TARGETPORT"
+    type: "int"
+    kind: "port"
+    description: "the port the Service uses to make the application accessible from outside the cluster"
```
Comment (bfoley13) [inline template/manifests/Service/draft.yaml:28]: TARGETPORT and PORT descriptions should swap. The target port is the port the application exposes while the port is the one on the service

File: template/manifests/Service/service.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{.APPNAME}}
+  labels:
+    app.kubernetes.io/name: {{.APPNAME}}
+    app.kubernetes.io/part-of: {{.PARTOF}}
+    kubernetes.azure.com/generator: {{.GENERATORLABEL}}
+spec:
+  type: ClusterIP
```
Comment (OliverMKing) [inline template/manifests/Service/service.yaml:10]: should type be configurable?

Comment (OliverMKing) [reply template/manifests/Service/service.yaml:10]: if type is configurable annotations maybe should be too for lb annotations (private, ip, etc)

Comment (davidgamero) [reply template/manifests/Service/service.yaml:10]: you mean allowing passing a string>string map of annotations as an input and then appending them in the template right?

Comment (OliverMKing) [reply template/manifests/Service/service.yaml:10]: i don't know what the draft format should be but if type is configurable being able to pass in annotations in some form is really useful

Comment (bfoley13) [reply template/manifests/Service/service.yaml:10]: I think having that configuration will come a little later, this was just meant to be a simple clusterIP service to begin with!

**Review by OliverMKing — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**